mlflow/utils/autologging_utils.py

```diff
@@ -9,3 +11,62 @@ def try_mlflow_log(fn, *args, **kwargs):
         fn(*args, **kwargs)
     except Exception as e:  # pylint: disable=broad-except
         warnings.warn("Logging to MLflow failed: " + str(e), stacklevel=2)
+
+
+def param_logger(fn, args=[], kwargs=None, unlogged=[]):
+    """
+    Log parameters explicitly passed to a function.
+    :param fn: function whose parameters are to be logged
+    :param args: arguments explicitly passed into fn
+    :param kwargs: kwargs explicitly passed into fn
+    :param unlogged: parameters not to be logged
+    :return: None
+    """
+    # Names of all parameters for the function
+    all_param_names = inspect.getargspec(fn)[0]
+
+    # Default values of all parameters with default values. Has length of n, and corresponds
+    # to values of last n elements in all_param_names
+    all_default_values = inspect.getargspec(fn)[3]
+
+    # Checking if default values are present for logging. Known bug that getargspec will return an
```

Comment:
Ah interesting is this actually a known bug / if so do we have a link to the Python bug report?

Reply:
Actually, there might be a way around this. `keras` has its own `getargspec()` [seen here](https://github.com/keras-team/keras/blob/master/keras/utils/generic_utils.py#L246), so I'll try and see if we can get around not logging `fit_generator()` params.

Reply:
After testing, even the `keras` specific `getargspec()` returns the same output as `inspect.getargspec()`. Turns out it wasn't a bug that was logged anywhere, but this function is deprecated in Python 3 but the only solution for backwards compatibility with Python 2.